docs/modules/nifi/pages/usage_guide/security.adoc

```diff
@@ -89,3 +90,54 @@ With this authorization method, a single user has administrator capabilities.
 === LDAP
 
 The operator uses the https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#fileusergroupprovider[`FileUserGroupProvider`] and https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#fileaccesspolicyprovider[FileAccessPolicyProvider] to bind the LDAP user to the NiFi administrator group. This user is then able to create and modify groups and polices in the web interface. These changes local to the `Pod` running NiFi and are *not* persistent.
+
+[#encrypting-sensitive-properties]
+== Encrypting sensitive properties on disk
+
+Some flows require storing a sensitive property like a password or access token, which is then stored on disk.
+NiFi https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#nifi_sensitive_props_key[encrypts these properties] in flow files.
+
+The sensitive property encryption is configured using the properties `keySecret`, `autoGenerate` and `algorithm` in the {crd-docs}/nifi.stackable.tech/nificluster/v1alpha1/#spec-clusterConfig-sensitiveProperties[`spec.clusterConfig.sensitiveProperties` {external-link-icon}^] configuration section of the NifiCluster resource.
+
+The `keySecret` configures the name of the Secret object that holds the encryption key; it is required to specify a Secret name.
+The Secret needs to contain the key as a value to the `nifiSensitivePropsKey` key.
+You can either specify a key yourself - in which case the key needs to be at least 12 characters long - or just specify the name of the secret and set `autoGenerate` to `true` (the default is false).
+If `autoGenerate` is false and no Secret with the given name in `keySecret` is found, the operator will raise an error.
+
+The `algorithm` property configures the encryption algorithm used to encrypt the sensitive properties.
+Consult the {crd-docs}/nifi.stackable.tech/nificluster/v1alpha1/#spec-clusterConfig-sensitiveProperties-algorithm[reference documentation {external-link-icon}^] for a list of supported algorithms.
+
+=== Autogenerated key example
+
+Let the operator generate a Secret with the name `nifi-sensitive-property-key`:
+
+[source,yaml]
+----
+sensitiveProperties:
+  keySecret: nifi-sensitive-property-key
+  autoGenerate: true
+----
+
+=== Custom key and encryption algorithm example
+
+Create the Secret yourself:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: nifi-sensitive-properties-key
+type: Opaque
```

Comment:
I have to admit I don't exactly know what this field is doing. If not required I would therefore prefer to remove it

Reply:
Ah, I think Opaque is just the default. I'm fine with removing it as well